Cutelyst/Plugins/Utils/Sql/sql.cpp

```diff
@@ -94,6 +94,34 @@ QVariantList Sql::queryToMapList(QSqlQuery &query)
     return ret;
 }
 
+QVariantMap Sql::queryToMap(QSqlQuery &query, const QString &key)
+{
+    QVariantMap ret;
+    QSqlRecord record = query.record();
+
+    if (!record.contains(key)) {
+      qCCritical(C_SQL) << "Field Name " << key << " not found in result set";
+      return ret;
+    }
+
+    int columns = record.count();
+    QStringList cols;
+
+    for (int i = 0; i < columns; ++i) {
+      cols.append(record.fieldName(i));
+    }
+
+    while (query.next()) {
+      QVariantHash h;
```

Comment:
Yes, but that is the "inner" Hash, where order does not matter.  The "outer" Map is sorted by key:
```
QVariantMap
|
+-- key "name": "a_project"
|   QVariantHash
|   |
|   +-- title: "Project A"
|   +-- description: "Description of Project A"
|
+-- key "name": "b_project"
|   QVariantHash
.   |
.   +-- title: "Project B"
.   +-- description: "Description of Project B"
.
```
I think from performance point of view we could leave the inner a Hash, the outer a Map.
What do you think?